src/query/storages/fuse/src/io/read/meta/segment_reader.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use std::io::Cursor;
+use std::io::Read;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use common_exception::Result;
+use futures::AsyncRead;
+use futures_util::AsyncReadExt;
+use serde::de::DeserializeOwned;
+use storages_common_table_meta::meta::decode;
+use storages_common_table_meta::meta::decompress;
+use storages_common_table_meta::meta::BlockMeta;
+use storages_common_table_meta::meta::Encoding;
+use storages_common_table_meta::meta::SegmentCompression;
+use storages_common_table_meta::meta::SegmentInfo;
+use storages_common_table_meta::meta::Statistics;
+use storages_common_table_meta::meta::Versioned;
+
+fn read_u64_exact<R>(reader: &mut R) -> Result<u64>
+where R: Read + Unpin + Send {
+    let mut buffer = [0; 8];
+    reader.read_exact(&mut buffer)?;
+    Ok(u64::from_le_bytes(buffer))
+}
+
+fn read_and_deserialize<R, T>(
+    reader: &mut R,
+    size: u64,
+    encoding: &Encoding,
+    compression: &SegmentCompression,
+) -> Result<T>
+where
+    R: Read + Unpin + Send,
+    T: DeserializeOwned,
+{
+    let mut compressed_data = vec![0; size as usize];
+    reader.read_exact(&mut compressed_data)?;
+
+    let decompressed_data = decompress(compression, compressed_data)?;
+
+    decode(encoding, &decompressed_data)
+}
+
+/// Reads a segment header from a binary stream and returns a `SegmentInfo` object.
+///
+/// This function reads the following fields from the stream and constructs a `SegmentInfo` object:
+///
+/// * `version` (u64): The version number of the segment.
```

Comment:
`version` could be `u16` and  `encoding`, `compression` could be `u8` I think.

Let's make these as small as possible.

Reply:
OK, I will modify!